src/lambda/commands/createNewSamApp.ts

```diff
@@ -206,6 +207,20 @@ export async function createNewSamApplication(
             return
         }
 
+        // Needs to be done or else gopls won't start
+        if (goRuntimes.includes(createRuntime)) {
+            try {
+                execSync('go mod tidy', { cwd: path.join(path.dirname(uri.fsPath), 'hello-world') })
+            } catch (err) {
+                getLogger().warn(
+                    localize(
+                        'AWS.message.warning.gotidyfailed',
+                        'Failed to initialize package directory with "go mod tidy". Launch config will not be automatically created.'
+                    )
```

Comment:
we have a similar message here: https://github.com/aws/aws-toolkit-vscode/blob/0507ee52790daeab1d8578e7e346f280ec98ed6c/src/lambda/commands/createNewSamApp.ts#L227-L240

I think we should show the identical message, for consistency, and also:

1. log the error
1. (in both cases) use the `showErrorWithLogs()` function
2. modify `showErrorWithLogs()` to optionally use `showWarningMessage` instead of `showErrorMessage`
    - (`showErrorWithLogs` should maybe be renamed later but don't worry about it now) 
3. modify `showErrorWithLogs()` to optionally also show the `Get Help...` button

Reply:
Agreed, but I'll do this in a separate PR since this one's already too big.